docs/query_engine_usage.md

```diff
@@ -349,3 +349,386 @@ BasicQuery(query_plan=query_plan)\
+      <td>-50000.0</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>chr1</td>
+      <td>chr1</td>
+      <td>chr1</td>
+      <td>chr1</td>
+      <td>-100000.0</td>
+      <td>-100000.0</td>
+      <td>0.0</td>
+    </tr>
+    <tr>
+      <th>3</th>
+      <td>chr1</td>
+      <td>chr1</td>
+      <td>chr1</td>
+      <td>chr1</td>
+      <td>-100000.0</td>
+      <td>-100000.0</td>
+      <td>50000.0</td>
+    </tr>
+    <tr>
+      <th>4</th>
+      <td>chr1</td>
+      <td>chr1</td>
+      <td>chr1</td>
+      <td>chr1</td>
+      <td>-100000.0</td>
+      <td>-100000.0</td>
+      <td>100000.0</td>
+    </tr>
+    <tr>
+      <th>...</th>
+      <td>...</td>
+      <td>...</td>
+      <td>...</td>
+      <td>...</td>
+      <td>...</td>
+      <td>...</td>
+      <td>...</td>
+    </tr>
+    <tr>
+      <th>245</th>
+      <td>chr2</td>
+      <td>chr2</td>
+      <td>chr2</td>
+      <td>chr2</td>
+      <td>100000.0</td>
+      <td>100000.0</td>
+      <td>-100000.0</td>
+    </tr>
+    <tr>
+      <th>246</th>
+      <td>chr2</td>
+      <td>chr2</td>
+      <td>chr2</td>
+      <td>chr2</td>
+      <td>100000.0</td>
+      <td>100000.0</td>
+      <td>-50000.0</td>
+    </tr>
+    <tr>
+      <th>247</th>
+      <td>chr2</td>
+      <td>chr2</td>
+      <td>chr2</td>
+      <td>chr2</td>
+      <td>100000.0</td>
+      <td>100000.0</td>
+      <td>0.0</td>
+    </tr>
+    <tr>
+      <th>248</th>
+      <td>chr2</td>
+      <td>chr2</td>
+      <td>chr2</td>
+      <td>chr2</td>
+      <td>100000.0</td>
+      <td>100000.0</td>
+      <td>50000.0</td>
+    </tr>
+    <tr>
+      <th>249</th>
+      <td>chr2</td>
+      <td>chr2</td>
+      <td>chr2</td>
+      <td>chr2</td>
+      <td>100000.0</td>
+      <td>100000.0</td>
+      <td>100000.0</td>
+    </tr>
+  </tbody>
+</table>
+<p>250 rows × 7 columns</p>
+</div>
+
+
+
+## Aggregating genomic data based on it's offset to a target region
+In this example, we extend the above use-case to aggregate the results based on the offset columns added. This is a common use-case to calculate aggregate statistics for different offset levels. To achieve this, we employ the same query plan as above and extend it using the `OffsetAggregation` query step.
+
+
+```python
+from spoc.query_engine import OffsetAggregation
+```
+
+The `OffsetAggregation` class requires the following parameters:
+- `value_columns`: Thie specifies the value to aggregate
+- `function`: The aggregation function to use. This is the enumerated type `AggregationFunction`
+- `densify_output`: Whether missing offset values should be filled with empty values (specific empty value depends on the aggregation function)
+
+Note that there are two different average functions available, `AVG` and `AVG_WITH_EMPTY`. `AVG` performs and average over all available columns, where as `AVG_WITH_EMPTY` counts missing offsets per regions as 0.
+
+
+```python
+query_plan = [
+    Overlap(target_regions, anchor_mode=Anchor(mode="ANY")),
```

Comment:
I would use `mode="ALL"` for pileups. Besides, it would be a nice opportunity to show case this mode.